neuralpredictors/layers/encoders/encoder3d.py

```diff
@@ -12,14 +12,30 @@ def __init__(self, core, readout, readout_nonlinearity, elu_xshift, elu_yshift):
         else:
             self.nonlinearity = core.nonlinearities[readout_nonlinearity]()
 
-    def forward(self, x, data_key=None):
+    def forward(self, x, data_key=None, pupil_center=None, trial_idx=None, shift=None, detach_core=False, **kwargs):
         out_core = self.core(x)
+        if detach_core:
+            x = x.detach()
+
+        if self.shifter:
+            if pupil_center is None:
+                raise ValueError("pupil_center is not given")
+            if shift is None:
+                time_points = x.shape[1]
+                pupil_center = pupil_center[:, :, -time_points:]
+                pupil_center = torch.transpose(pupil_center, 1, 2)
+                pupil_center = pupil_center.reshape(((-1,) + pupil_center.size()[2:]))
+                shift = self.shifter[data_key](pupil_center, trial_idx)
```

Comment:
but I guess `shift` and `shifter` are also slightly semantically different

A I get it `shift` if the output of the `shifter` and theoretically it could be provided for the readout from somewhere else, not from the `model.shifter` (that the only reason why this parameter is here and I also tried to make it consistency with the 2d core)